qiskit/quantum_info/operators/operator.py

```diff
@@ -615,6 +615,62 @@ def _multiply(self, other):
         ret._data = other * self._data
         return ret
 
+    def _equal_with_ancillas(
+        self,
+        other: Operator,
+        ancilla_qubits: list,
+        ignore_phase: bool = False,
+        rtol: float | None = None,
+        atol: float | None = None,
+    ) -> bool:
+        r"""Test if two Operators are equal on the subspace where ancilla qubits
+        are :math:`|0\rangle`.
+
+        Args:
+            other (Operator): an operator object.
+            ancilla_qubits: a list of clean ancilla qubits.
+            rtol (float): relative tolerance value for comparison.
+            atol (float): absolute tolerance value for comparison.
+            ignore_phase (bool): ignore complex-phase difference between matrices.
+
+        Returns:
+            bool: True if operators are equal up to clean ancilla qubits.
+        """
+        if self.dim != other.dim:
```

Comment:
Add type checking on `other` similarly to how it's done in `equiv()`

Reply:
The `equiv` code does more than just type-checking: if converts `other` to an `Operator` when it's not already, so for example one can check the equivalence of an `Operator` and a `QuantumCircuit`. I don't think this fully makes sense for this (currently private) function.